porcupine/pluginloader.py

```diff
@@ -35,95 +35,71 @@
     DISABLED_BY_SETTINGS = enum.auto()
+
+    # The plugin wasn't loaded because it was listed in a `--without-plugins`
+    # argument given to Porcupine.
     DISABLED_ON_COMMAND_LINE = enum.auto()
+
+    # Importing the plugin raised an error.
     IMPORT_FAILED = enum.auto()
+
+    # The plugin was imported successfully, but its `setup()` function
+    # raised an exception or logged an error.
+    #
+    # In a plugin named `foo`, any message logged with severity `ERROR`
+    # or `CRITICAL` to the logger named `porcupine.plugins.foo` counts as
+    # logging an error. Therefore you can do this::
+    #
+    #    import logging
+    #
+    #    log = logging.getLogger(__name__)  # __name__ == "porcupine.plugins.foo"
+    #
+    #    def setup() -> None:
+    #        if bar_is_installed:
+    #            ...
+    #        else:
+    #            log.error("bar is not installed")
+    #
+    # When bar is not installed, this plugin will show a one-line error
+    # message in the plugin manager and the terminal. If an exception is
+    # raised, the full traceback is shown instead.
     SETUP_FAILED = enum.auto()
+
+    # Plugins with this status were imported, but their `setup_before` and
+    # `setup_after` lists make it impossible to determine the correct order
+    # for calling their `setup()` function.
+    #
+    # For example, if plugin *A* should be set up before *B*, *B* should be
+    # set up before *C* and *C* should be set up before *A*, then *A*, *B*
+    # and *C* will all fail to load with this status.
     CIRCULAR_DEPENDENCY_ERROR = enum.auto()
 
 
 @dataclasses.dataclass(eq=False)
 class PluginInfo:
     """
-    This :mod:`dataclass <dataclasses>` represents a plugin.
-
-    It's usually better to use ``info.setup_before``
-    instead of accessing ``info.module.setup_before`` directly.
-    Not all plugins define a ``setup_before`` variable, and if it's not present,
-    then ``info.setup_before`` is an empty set.
-    This also applies to ``setup_after``.
-
-    The value of *error* depends on *status*:
-
-        * If *status* is ``LOADING``, ``ACTIVE``, ``DISABLED_BY_SETTINGS`` or
-          ``DISABLED_ON_COMMAND_LINE``, then *error* is ``None``.
-        * If *status* is ``IMPORT_FAILED`` or ``SETUP_FAILED``, then *error*
-          is a Python error message, starting with
-          ``Traceback (most recent call last):``.
-        * If *status* is ``CIRCULAR_DEPENDENCY_ERROR``, then *error* is a
-          user-readable one-line message.
+    This dataclass represents a plugin.
+
+    The value of `error` depends on `status`:
+    - `LOADING`: `error` is `None`
+    - `ACTIVE`: `error` is `None`
+    - `DISABLED_BY_SETTINGS`: `error` is `None`
+    - `DISABLED_ON_COMMAND_LINE`: `error` is `None`
+    - `IMPORT_FAILED`: `error` is a Python error message
+    - `SETUP_FAILED`: `error` is a Python error message
+    - `CIRCULAR_DEPENDENCY_ERROR`: `error` is a user-readable one-line message
```

Comment:
This should be moved just above the `error` field, and replaced with something shorter, like this

```
    The value of `error` is:
    - a user-readable one-line message, if `status` is `CIRCULAR_DEPENDENCY_ERROR`
    - a Python error message, if `status` is `IMPORT_FAILED` or `SETUP_FAILED`
    - `None` otherwise.
```